hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -689,6 +707,7 @@ public String getBucketLocation() throws IOException {
    * @return the region in which a bucket is located
    * @throws IOException on any failure.
    */
+  @VisibleForTesting
```

Comment:
... I could kinda see clients wanting to use this. I know for HBOSS I almost did when I was toying with a potential DynamoDB locking implementation..

Reply:
The new storecontext API exports this as an on-demand operation.: `fs.createStoreContext().getBucketLocation()`